lib/nats.js

```diff
@@ -886,6 +886,7 @@ Client.prototype.sendConnect = function () {
  * @api private
  */
 Client.prototype.createConnection = function () {
+  this.emit('dialing')
```

Comment:
We freeform these all the time or are they constants somewhere?

Reply:
these are only for testing - we document all the ones that are official configuration/event properties.

Reply:
But these will be in production too and users can start to depend on them yes?

Reply:
we are not responsible for users digging into the source and making dependencies on non-public features. This is simply used by a test to confirm that a dial took place when it is aborted by a tls certificate error.

Reply:
Understood but they can, lets just make it a constant. Don't have to document it.

Reply:
Would there be a way to use existing connection field that are updated in this process, for instance `reconnects` (I understand that this test fails during initial connect, but wonder if node is using the same logic than if it was reconnecting even on initial failed connect).

Reply:
yes hooking on `reconnecting` works perfect. - will add constants for events in a different PR. this way we can group them under `Events` and `TestEvents` to further document the intention..